core/test/vault-test.ts

```diff
@@ -0,0 +1,381 @@
+import { fetchLiquidationRatioAndOracleAddress, fetchVault, getVaultTransaction, liquidateVault } from '../src/vaults';
+import { getOsmPrices } from '../src/oracles';
+import { resetNetwork } from '../helpers/hardhat';
+import { setupRpcUrlAndGetNetworks } from '../src/rpc';
+import { HARDHAT_PRIVATE_KEY, HARDHAT_PUBLIC_KEY, LOCAL_RPC_URL, TEST_NETWORK } from '../helpers/constants';
+import { expect } from 'chai';
+import { VaultTransactionLiquidated, VaultTransactionNotLiquidated, Vault, CollateralType } from '../src/types';
+import BigNumber from '../src/bignumber';
+import { createWalletFromPrivateKey } from '../src/signer';
+import COLLATERALS from '../src/constants/COLLATERALS';
+
+const compareVaultTransactionsNotLiquidated = (
+    expected: VaultTransactionNotLiquidated,
+    actual: VaultTransactionNotLiquidated
+) => {
+    expect(expected.state).to.eq(actual.state);
+    expect(expected.collateralType).to.eq(actual.collateralType);
+    expect(expected.address).to.eq(actual.address);
+    expect(expected.network).to.eq(actual.network);
+    expect(expected.nextPriceChange?.toISOString()).to.eq(actual.nextPriceChange?.toISOString());
```

Comment:
Would [`deep-equal-in-any-order`](https://www.chaijs.com/plugins/deep-equal-in-any-order/) help here?

Reply:
there're other fields such as `lastSyncedAt` that are skipped. Also it would heavily influence the debugging process: comparing as it is done rn allows to pre-print the `BigNumber.toFixed()` to spot the exact expected value.

Reply:
In general, I expect some kind of helper that utilises `json pointer` to just list which values should be compared can add more clarity here and also help with debugging